src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -231,6 +243,8 @@ def __post_init__(self) -> None:
                 self.initial_buffer_len,
                 self.warn_buffer_len,
             )
+        if self.align_to is not None and self.align_to.tzinfo is None:
+            raise ValueError(f"align_to ({self.align_to}) should be an aware datetime")
```

Comment:
Might be useful in other places too to have that. Maybe make it a function in the future? Not in this PR.

Reply:
Maybe, I'm not sure, if the idea is to rely on `pylint` for this, it might be redundant at the end. To be honest I wasn't sure if I should include it not, littering all the code with checks for datetime awareness also seems a bit overkill.

I'm really undecided about this, also while converting `int` in seconds to `timedelta` I was thinking it might be really useful to have a `seconds(x)` alias for `timedelta(seconds=x)`. Or to have some sort of `now()` that produces a timezone aware UTC now.

But then I think it is just saving a few characters and it does make the code a bit less obvious for the casual reader (anyone familiar with python but not with our code knows what `timedelta(seconds=x)` is, but `seconds(x)` is an `int`? a `float`? a `timedelta`?)